velox/vector/VariantToVector.cpp

```diff
@@ -50,8 +59,21 @@ ArrayVectorPtr variantArrayToVectorImpl(
   for (vector_size_t i = 0; i < variantArraySize; i++) {
     if (!variantArray[i].isNull()) {
       // `getOwnedValue` copies the content to its internal buffers (in case of
-      // string/StringView); no-op for other primitive types.
-      arrayElements->set(i, T(variantArray[i].value<KIND>()));
+      // string/StringView). DECIMAL variant arrays require the optional
+      // parameter in the DecimalCapsule, unscaledValue, to be set in order to
+      // update arrayElements in the returned ArrayVector with values of
+      // NativeType (UnscaledShortDecimal/UnscaledLongDecimal).
+      if constexpr (isDecimalType) {
```

Comment:
If the DecimalCapsule value is missing, it means the value is null.
See the definition of variant::isNull().
This specialization is not required as well.

Reply:
Got it, updated to read the values as NULL. Removing the specialization results in the following compile errors though:
```
No matching conversion for functional-style cast from 'const typename detail::VariantTypeTraits<(TypeKind)'\f'>::stored_type' (aka 'const DecimalCapsule<facebook::velox::UnscaledShortDecimal>') to 'T' (aka 'facebook::velox::UnscaledShortDecimal')
No matching conversion for functional-style cast from 'const typename detail::VariantTypeTraits<(TypeKind)'\r'>::stored_type' (aka 'const DecimalCapsule<facebook::velox::UnscaledLongDecimal>') to 'T' (aka 'facebook::velox::UnscaledLongDecimal')
```

Reply:
The check above `if (!variantArray[i].isNull())` will handle null. There is no need to address it here.
You simplify as
```
if constexpr (KIND == TypeKind::SHORT_DECIMAL || KIND == TypeKind::LONG_DECIMAL) {
   arrayElements->set(i, T(variantArray[i].value<KIND>().value()));
} else {
  arrayElements->set(i, T(variantArray[i].value<KIND>()));
}
```

Reply:
Got it, thanks!